rs/nns/governance/src/neuron_store.rs

```diff
@@ -792,14 +840,54 @@ impl NeuronStore {
         self.filter_map_active_neurons(filter, |n| n.id())
     }
 
-    /// Returns an iterator of all voting-eligible neurons
-    pub fn voting_eligible_neurons(&self, now_seconds: u64) -> impl Iterator<Item = &Neuron> {
-        // This should be safe to do without with_neuron because
-        // all voting_eligible neurons should be in the heap
-        self.active_neurons_iter().filter(move |&neuron| {
-            neuron.dissolve_delay_seconds(now_seconds)
-                >= MIN_DISSOLVE_DELAY_FOR_VOTE_ELIGIBILITY_SECONDS
-        })
+    pub fn create_ballots_for_standard_proposal(
+        &self,
+        now_seconds: u64,
+    ) -> (
+        HashMap<u64, Ballot>,
+        u128, /*deciding_voting_power*/
+        u128, /*potential_voting_power*/
+    ) {
+        let mut ballots = HashMap::<u64, Ballot>::new();
+        let mut deciding_voting_power: u128 = 0;
+        let mut potential_voting_power: u128 = 0;
+
+        let mut process_neuron = |neuron: Neuron| {
+            if neuron.is_inactive(now_seconds)
+                || neuron.dissolve_delay_seconds(now_seconds)
+                    < MIN_DISSOLVE_DELAY_FOR_VOTE_ELIGIBILITY_SECONDS
+            {
+                return;
+            }
+
+            let voting_power = neuron.deciding_voting_power(now_seconds);
+            deciding_voting_power += voting_power as u128;
+            potential_voting_power += neuron.potential_voting_power(now_seconds) as u128;
+            ballots.insert(
+                neuron.id().id,
+                Ballot {
+                    vote: Vote::Unspecified as i32,
+                    voting_power,
+                },
+            );
+        };
+
+        if self.use_stable_memory_for_all_neurons {
+            with_stable_neuron_store(|stable_store| {
+                // Not including unneeded sections gives us greater than 20x instructions improvement
+                for neuron in stable_store.range_neurons_sections(.., NeuronSections::default()) {
+                    process_neuron(neuron);
+                }
+            })
+        } else {
+            self.with_active_neurons_iter(|iter| {
```

Comment:
This function already makes the distinction, so does it need to be differentiated here too?